android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java

```diff
@@ -273,30 +273,36 @@ public boolean handleMessage(Message msg)
 	public KrollDict getRect()
 	// clang-format on
 	{
-		KrollDict d = new KrollDict();
+		View v = null;
 		if (view != null) {
-			View v = view.getOuterView();
-			if (v != null) {
-				int position[] = new int[2];
-				v.getLocationInWindow(position);
-
-				TiDimension nativeWidth = new TiDimension(v.getWidth(), TiDimension.TYPE_WIDTH);
-				TiDimension nativeHeight = new TiDimension(v.getHeight(), TiDimension.TYPE_HEIGHT);
-				TiDimension nativeLeft = new TiDimension(position[0], TiDimension.TYPE_LEFT);
-				TiDimension nativeTop = new TiDimension(position[1], TiDimension.TYPE_TOP);
-				TiDimension localLeft = new TiDimension(v.getX(), TiDimension.TYPE_LEFT);
-				TiDimension localTop = new TiDimension(v.getY(), TiDimension.TYPE_TOP);
+			v = view.getOuterView();
+		}
+		return getViewRect(v);
+	}
 
-				// TiDimension needs a view to grab the window manager, so we'll just use the decorview of the current window
-				View decorView = TiApplication.getAppRootOrCurrentActivity().getWindow().getDecorView();
-				if (decorView != null) {
-					d.put(TiC.PROPERTY_WIDTH, nativeWidth.getAsDefault(decorView));
-					d.put(TiC.PROPERTY_HEIGHT, nativeHeight.getAsDefault(decorView));
-					d.put(TiC.PROPERTY_X, localLeft.getAsDefault(decorView));
-					d.put(TiC.PROPERTY_Y, localTop.getAsDefault(decorView));
-					d.put(TiC.PROPERTY_X_ABSOLUTE, nativeLeft.getAsDefault(decorView));
-					d.put(TiC.PROPERTY_Y_ABSOLUTE, nativeTop.getAsDefault(decorView));
-				}
+	protected KrollDict getViewRect(View v)
+	{
+		KrollDict d = new KrollDict();
+		if (v != null) {
+			int position[] = new int[2];
+			v.getLocationInWindow(position);
+
+			TiDimension nativeWidth = new TiDimension(v.getWidth(), TiDimension.TYPE_WIDTH);
+			TiDimension nativeHeight = new TiDimension(v.getHeight(), TiDimension.TYPE_HEIGHT);
+			TiDimension nativeLeft = new TiDimension(position[0], TiDimension.TYPE_LEFT);
+			TiDimension nativeTop = new TiDimension(position[1], TiDimension.TYPE_TOP);
+			TiDimension localLeft = new TiDimension(v.getX(), TiDimension.TYPE_LEFT);
+			TiDimension localTop = new TiDimension(v.getY(), TiDimension.TYPE_TOP);
+
+			// TiDimension needs a view to grab the window manager, so we'll just use the decorview of the current window
+			View decorView = TiApplication.getAppRootOrCurrentActivity().getWindow().getDecorView();
+			if (decorView != null) {
+				d.put(TiC.PROPERTY_WIDTH, nativeWidth.getAsDefault(decorView));
+				d.put(TiC.PROPERTY_HEIGHT, nativeHeight.getAsDefault(decorView));
+				d.put(TiC.PROPERTY_X, localLeft.getAsDefault(decorView));
+				d.put(TiC.PROPERTY_Y, localTop.getAsDefault(decorView));
+				d.put(TiC.PROPERTY_X_ABSOLUTE, nativeLeft.getAsDefault(decorView));
+				d.put(TiC.PROPERTY_Y_ABSOLUTE, nativeTop.getAsDefault(decorView));
```

Comment:
We actually don't need the DecorView here. We can pass in the `v` variable as an argument to `getAsDefault()` and will get the same results.

I know you didn't make it this way. That's how the old code worked. Now that I look at it, the `TiApplication.getAppRootOrCurrentActivity().getWindow()` part will crash if Developer Option "Don't keep activities" is enabled and the app is backgrounded. The `getAsDefault()` method simply wants a `Context` to fetch `DisplayMetrics` from and we can get that from any `View`.